File: code-coaches/medium/youtube-link-finder/YouTubeLinkFinder.java
import java.util.Scanner;

public class YouTubeLinkFinder {

    public static String extractVideoId(String url) throws IllegalArgumentException {
        if (url.contains("youtube.com/watch")) {
            String[] parts = url.split("v=");
            if (parts.length == 2) {
                return parts[1];
            }
        } else if (url.contains("youtu.be/")) {
            String[] parts = url.split("/");
            if (parts.length > 0) {
                return parts[parts.length - 1];
            }
        }
        throw new IllegalArgumentException("Invalid YouTube video URL");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String url = scanner.nextLine();

        try {
            String videoId = extractVideoId(url);
            System.out.println(videoId);
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }

        scanner.close();
    }
}

File: code-coaches/medium/youtube-link-finder/YouTubeLinkFinder.cs
using System;

class YouTubeLinkFinder
{
    static string ExtractVideoId(string url)
    {
        if (url.Contains("youtube.com/watch"))
        {
            string[] parts = url.Split(new string[] { "v=" }, StringSplitOptions.None);
            if (parts.Length == 2)
            {
                return parts[1];
            }
        }
        else if (url.Contains("youtu.be/"))
        {
            string[] parts = url.Split('/');
            if (parts.Length > 0)
            {
                return parts[parts.Length - 1];
            }
        }
        throw new ArgumentException("Invalid YouTube video URL");
    }

    static void Main(string[] args)
    {
        string url = Console.ReadLine();
        string videoId = ExtractVideoId(url);
        Console.WriteLine(videoId);
    }
}
